File: frontend/src/api/jobs.ts
import { apiGet } from "./client";
import type { JobListResponse } from "../types";

export type JobFilters = {
  search?: string;
  location?: string;
  remote?: string;
};

export function getJobs(filters: JobFilters = {}): Promise<JobListResponse> {
  const params = new URLSearchParams();
  if (filters.search) params.set("search", filters.search);
  if (filters.location) params.set("location", filters.location);
  if (filters.remote) params.set("remote", filters.remote);

  const query = params.toString();
  return apiGet<JobListResponse>(`/api/jobs${query ? `?${query}` : ""}`);
}

File: backend/app/routers/jobs.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.database import get_db
from app.schemas.job import JobListResponse, JobOut
from app.services import job_service

router = APIRouter(prefix="/api/jobs", tags=["jobs"])


@router.get("", response_model=JobListResponse)
def list_jobs(
    search: str | None = None,
    location: str | None = None,
    remote: str | None = None,
    limit: int = Query(20, le=100),
    offset: int = 0,
    db: Session = Depends(get_db),
):
    results, total = job_service.list_jobs(db, search, location, remote, limit, offset)
    return JobListResponse(
        count=total,
        results=[JobOut.model_validate(job) for job in results],
    )